docs/docs/adding-comments.md

```diff
@@ -0,0 +1,85 @@
+
+But in this guide, we'll assume you want something fast with as little manual configuration and setup as possible. Disqus has several things going for it in this regard:
+
+- It [seems to be by far the most widely used service](https://www.datanyze.com/market-share/comment-systems/disqus-market-share).
+- It is low maintenance, meaning [moderating your comments and maintaining your forum](https://help.disqus.com/moderation/moderating-101) is easy.
+- It provides official [React support](https://github.com/disqus/disqus-react).
+- It offers a [generous free tier](https://disqus.com/pricing).
+- It’s easy to comment: Disqus has a large existing user base and the onboarding experience for new users is fast since you can register with your Google, Facebook or Twitter account. You can also easily share your review about the post through those channels.
+- Its commenting interface has a distinct but unobtrusive look that many users will instantly recognize and trust.
+- All Disqus components are lazy-loaded, meaning they won't negatively impact the load times of your posts.
+
+## Implementing Disqus
+
+![Disqus logo](images/disqus-logo.svg)
+
+Here are the steps for adding Disqus comments to your own blog:
+
+1. [Sign-up to Disqus](https://disqus.com/profile/signup). During the process you'll have to choose a shortname for your site. This is how Disqus will identify comments coming from your site. Copy that for later.
+2. Install the Disqus React package
+
+   ```sh
+   yarn add disqus-react
+   ```
+
+3. Add the shortname from step 1 as something like `GATSBY_DISQUS_NAME` to your `.env` and `.env.example` files so that people forking your repo will know that they need to supply this value to get comments to work. (You need to prefix the environment variable with `GATSBY_` in order to [make it available to client side code](https://www.gatsbyjs.org/docs/environment-variables/#client-side-javascript).)
+   ```:envtitle=.env.example
+   ...
+   # enables Disqus comments for blog posts
+   GATSBY_DISQUS_NAME=insertValue
+   ```
+   ```:envtitle=.env
+   ...
+   GATSBY_DISQUS_NAME=yourOwnSiteShortname
+   ```
+4. In your blog post template (usually `src/templates/post.js`) import the `DiscussionEmbed` React component.
+
+   ```js{3}:title=src/templates/post.js
+   import React from 'react'
+   import { graphql } from 'gatsby'
+   import { DiscussionEmbed } from 'disqus-react'
+   ...
+   ```
+
+   Then define your Disqus configuration object
+
+   ```js
+   const disqusConfig = {
+     shortname: process.env.GATSBY_DISQUS_NAME,
+     config: { identifier: slug, title },
+   }
+   ```
+
+   where `identifier` must be a string or number that uniquely identifies the post. It could be the post's slug, title or some ID. Finally, add `DiscussionEmbed` to the JSX of your post template.
+
+   ```jsx{6}:title=src/templates/post.js
```

Comment:
Same idea here, want to use `highlight-line` instead? It's a lot easier to maintain if we make changes to the snippets in the future.

```suggestion
   ```jsx:title=src/templates/post.js
```